homework/index.js

```diff
@@ -30,13 +30,162 @@
     return elem;
   }
 
+  function defaultSelectedRepository(data, selectedText) {
+    let elem;
+    data.filter(element => {
+      if (element.name === selectedText) {
+        elem = element;
+      }
+      return elem;
+    });
+    return elem;
+  }
+
+  function buildHtml(data, root) {
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('p', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header, {
+      class: 'repo-selector',
+      arialabel: 'HYF Repositories',
+    });
+
+    data
+      .map(repository => repository.name)
+      .sort((a, b) => a.localeCompare(b, { caseFirst: 'lower' }))
+      .map((element, index) => createAndAppend('option', select, { value: index, text: element }));
+
+    const container = createAndAppend('div', root, { id: 'container' });
+    const leftDiv = createAndAppend('div', container, { class: 'left-div whiteframe' });
+    const table = createAndAppend('table', leftDiv);
+    const tbody = createAndAppend('tbody', table, { id: 'tbody' });
+
+    const defaultRepo = defaultSelectedRepository(data, select[select.selectedIndex].text);
+
+    function showSelectedConts(dataCont) {
```

Comment:
You should avoid nested functions like this. The reason is that nested function obscure where the data comes from that they need. A nested function has access to the data of its outer function. In this case, the `showSelectedConst` function accesses the `container` variable from its outer function, but that you would have to look hard to understand that. If you place this function outside of the `buildHtml` function you immediately see the `container` variable being red underlined in VSCode, indicating that it is now not defined. To solve that, you can pass it through the parameter list, making it absolutely clear what this function needs from the outside world to do its work:

```js
function showSelectedConts(dataCont, container) {
 ...
}

function buildHtml(data, root) {
  ...
  showSelectConts(contributors, container);
  ...
}
```

BTW, avoid unnecessary abbreviations like `Conts` in `showSelectedConts`. It is only a few more characters to make it `showSelectedContributors` and so much nicer to read without needing to guess.